src/extension.ts

```diff
@@ -486,6 +486,207 @@ export function activate(context: vscode.ExtensionContext) {
+            outputBuffers.push(new Buffer(bufferOffset - bufferData.length));
+        }
+
+        for (const accessorType of ['input', 'output']) {
+            const values = newData[accessorType];
+            const accessorComponents = accessorType == 'input' ? 1 : components;
+            const max = new Array(accessorComponents).fill(Number.NEGATIVE_INFINITY);
+            for (let i = 0; i < values.length; i++) {
+                const j = i % accessorComponents;
+                max[j] = Math.max(max[j], values[i]);
+            }
+            const min = new Array(accessorComponents).fill(Number.POSITIVE_INFINITY);
+            for (let i = 0; i < values.length; i++) {
+                const j = i % accessorComponents;
+                min[j] = Math.min(min[j], values[i]);
+            }
+            const float32Values = Float32Array.from(values);
+            const accessor = {
+                "bufferView": glTF.bufferViews.length,
+                "componentType": 5126,
+                "count": values.length / accessorComponents,
+                "type": accessorType == 'input' ? 'SCALAR' : samplerType,
+                "max": max,
+                "min": min
+            };
+            glTF.accessors[animationPointer.json[accessorType]] = accessor;
+            const newBufferView = {
+                "buffer": bufferView.buffer,
+                "byteOffset": bufferOffset,
+                "byteLength": float32Values.byteLength,
+            };
+            glTF.bufferViews.push(newBufferView);
+            if (alignedLength(float32Values.byteLength) != float32Values.byteLength) {
+                throw new Error('Float32Array not 4 byte length');
+            }
+            bufferOffset += float32Values.byteLength;
+            outputBuffers.push(Buffer.from(float32Values.buffer));
+        }
+
+        const createBufferFilename = !bufferJson.hasOwnProperty('uri') || bufferJson.uri.startsWith('data:');
+
+        let binFilename = bufferJson.uri;
+        if (createBufferFilename) {
+            let targetBasename = activeTextEditor.document.fileName;
+            if (path.extname(activeTextEditor.document.fileName).length > 1) {
+                const components = activeTextEditor.document.fileName.split('.');
+                components.pop();
+                targetBasename = components.join('.');
+            }
+            binFilename = targetBasename + '_data.bin';
+        }
+        bufferJson.uri = binFilename;
+
+        const dir = path.dirname(activeTextEditor.document.fileName);
+        const targetFilename = path.join(dir, binFilename);
+
+        const finalBuffer = Buffer.concat(outputBuffers);
+        fs.writeFileSync(targetFilename, finalBuffer, 'binary');
+
+        bufferJson.byteLength = finalBuffer.length;
+        const newJson = JSON.stringify(glTF, null, '  ');
+        await activeTextEditor.document.save();
+        fs.writeFileSync(activeTextEditor.document.fileName, newJson, 'utf8');
```

Comment:
Minor wrinkle, but if the 3D preview window is open when this happens, it doesn't appear to automatically update with the new changes.  I only tested once though, but I pressed Alt-o and it saved the file, and the 3D window didn't blink or reload.  I had to manually make another harmless change to the spacing and save that to force a reload.

Reply:
Messing with this some more.  If the document is un-saved with edits and you press alt-o, the act of saving causes a 3D preview reload, but it reloads the wrong version (the version being saved, with the edits still in the extras, not the exported version).  If the user does manually save some edits, the reload will be triggered then, and no reload will happen on alt-o.  Apparently neither case results in the preview window showing the newly written bin file.

Reply:
The latest changes don't save and just do in place editing. I've added a note in the readme suggesting to the customer to save when they are ready.